### PR TITLE
New function: get_matched_basenames()

### PR DESCRIPTION
**Pull request by hoechenberger**
PR Description
--------------

`path.get_matched_basenames()` allows users to retrieve a list of `BIDSPaths` matching specified entities. The entire dataset (BIDS root) will be globbed, and a list of matching `BIDSPaths` will be returned. In case of no match, returns an empty list.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

File: mne_bids/path.py [hunk 1]
```diff
@@ -941,3 +941,99 @@ def _path_to_str(var):
...
+    ses_str = f'_ses-{session}' if session else r'(|_ses-([^_]+))'
+    task_str = f'_task-{task}' if task else r'(|_task-([^_]+))'
+    acq_str = f'_acq-{acquisition}' if acquisition else r'(|_acq-([^_]+))'
+    run_str = f'_run-{run}' if run else r'(|_run-([^_]+))'
+    proc_str = f'_proc-{processing}' if processing else r'(|_proc-([^_]+))'
+    rec_str = f'_rec-{recording}' if recording else r'(|_rec-([^_]+))'
+    space_str = f'_space-{space}' if space else r'(|_space-([^_]+))'
+    split_str = f'_split-{split}' if split else r'(|_split-([^_]+))'
+    # XXX do we want MRI here?
+    kind_str = f'_{kind}' if kind else r'_(meg|eeg|ieeg|mri)'
```
Comment (agramfort) [inline mne_bids/path.py:960]: @hoechenberger shall we use ALLOWED_MODALITY_EXTENSIONS to build this string here?

Comment (hoechenberger) [reply mne_bids/path.py]: Yep, did that now

Comment (hoechenberger) [reply mne_bids/path.py]: Errm, using ALLOWED_FILENAME_KINDS, I mean.

**Comment by agramfort** [comment]
since this is not affecting any core feature and it's a blocker for another PR I'll just merge this and we can improve later.

thx @hoechenberger

**Pull request merged**